opossum-tool/OpossumUI
# Handle 'wasPreferred' for merged signals

### Summary of changes

- Added `wasPreferred` to the list of keys to be excluded when calculating the hash for the an attribution
- Sets `wasPreferred` to true for the `DisplayPackageInfo` if at least for one of the origin attributions the field `wasPreferred` was set to true
- Added tests


### Context and reason for change

Recently we have introduced the `wasPreferred` field  that allowes to highlight external Attributions that were previously preferred. 

In this diff we make sure that the `wasPreferred` filed is handled correctly during merging of multiple attributions.

### How can the changes be tested

- Checkout commit
- run `yarn start`
- delete `opossum_input.opossum` in `example-files` if pressent
- open `opossum_input.json`
- create a new file with a ".opossum" extension during opening process
- In audit view navigate to `/index.ts`
- Check that the merged signal with the license `Apache License Version 2.0` for `TypeScript 3.9.6` is marked by the gray star for `wasPreferred`
